staging/src/k8s.io/client-go/util/certificate/certificate_manager.go

```diff
@@ -95,7 +95,8 @@ type Config struct {
 	RequestedCertificateLifetime *time.Duration
 	// Usages is the types of usages that certificates generated by the manager
 	// can be used for.
-	Usages []certificates.KeyUsage
+	Usages    []certificates.KeyUsage
+	GetUsages func(privateKey interface{}) []certificates.KeyUsage
```

Comment:
add godoc for this and document it is mutually exclusive with `Usages`

Reply:
Done.